packages/browser/__tests__/Session.spec.ts

```diff
@@ -267,6 +267,158 @@ describe("Session", () => {
       );
       expect(clientAuthnHandleIncomingRedirect).toHaveBeenCalled();
     });
+
+    // This workaround will be removed after we've settled on an API that allows us to silently
+    // re-activate the user's session after a refresh:
+    describe("(using the temporary workaround for losing a session after refresh)", () => {
+      it("does not report the user to be logged in when the workaround cookie is not present, and just executes the redirect", async () => {
+        const clientAuthentication = mockClientAuthentication();
+        clientAuthentication.handleIncomingRedirect = jest.fn(
+          async (_url: string) => undefined
+        );
+        const mySession = new Session({ clientAuthentication });
+        await mySession.handleIncomingRedirect("https://some.url");
+        expect(mySession.info.isLoggedIn).toEqual(false);
+        expect(mySession.info.webId).toBeUndefined();
+        expect(
+          clientAuthentication.handleIncomingRedirect
+        ).toHaveBeenCalledTimes(1);
+      });
+
+      it("re-initialises the user's WebID without redirection if the proper cookie is set", async () => {
+        // Mock localStorage's `getItem` method:
+        const existingLocalStorage = window.localStorage;
+        const localStorageMock = {
+          getItem: jest.fn(() =>
+            JSON.stringify({
+              webId: "https://my.pod/profile#me",
+              sessions: {
+                "https://my.pod/": { expiration: 9000000000000 },
+              },
+            })
+          ),
+        };
+
+        Object.defineProperty(window, "localStorage", {
+          value: localStorageMock,
+          writable: true,
+        });
+        const clientAuthentication = mockClientAuthentication();
+        clientAuthentication.handleIncomingRedirect = jest.fn();
+        const mySession = new Session({ clientAuthentication });
+        await mySession.handleIncomingRedirect("https://some.url");
+        expect(mySession.info.isLoggedIn).toEqual(true);
+        expect(mySession.info.webId).toEqual("https://my.pod/profile#me");
+        expect(
+          clientAuthentication.handleIncomingRedirect
+        ).toHaveBeenCalledTimes(0);
+
+        // Remove the mocked method:
```

Comment:
Is this safe - i.e. if this one test fails mid-way, will it leave bad state? If so, this could be in a `try { } finally { <code here> }` block, or a Jest `afterTest()` cleanup function...?

Reply:
It had some side-effect, and an error thrown in the middle of this test failed at least one other test in a non-trivial manner. I added a `try/finally` here: 17d07888e28a48e8dca085b94e9571d9927d4d21